File: src/Dto/Project.Dto.Http/Post/PostDto.cs
namespace Project.Dto.Http.Post;

public class PostDto
{
    public PostDto(Guid id, string title, decimal salary, Guid companyId, bool isDeleted)
    {
        Id = id;
        Title = title;
        Salary = salary;
        CompanyId = companyId;
        IsDeleted = isDeleted;
    }

    public Guid Id { get; set; }
    public string Title { get; set; }
    public decimal Salary { get; set; }
    public Guid CompanyId { get; set; }
    public bool IsDeleted { get; set; }
}

File: front/src/app/dto/post.dto.ts
/**
 * DTO for post received from API
 */
export interface PostDto {
  id: string;
  title: string;
  salary: number;
  companyId: string;
  isDeleted: boolean;
}

/**
 * DTO for creating a post
 */
export interface CreatePostDto {
  title: string;
  salary: number;
  companyId: string;
}

/**
 * DTO for updating a post
 */
export interface UpdatePostDto {
  title?: string;
  salary?: number;
}

